cpp/JobSystem.hpp

```diff
@@ -0,0 +1,175 @@
+#pragma once
+
+#include "Thread.hpp"
+#include "Functional.hpp"
+#include "Vector.hpp"
+#include "MPMCQueue.hpp"
+
+#include <atomic>
+
+namespace hsd
+{
+    struct Job;
+
+    using JobFn = void(*)(Job);
+    using Counter = std::atomic<size_t>;
+
+    struct Job
+    {
+        JobFn task;
+        void** data = nullptr;
+    };
+
+/*
+    template<u8 numargs, typename T>
+    struct JobData
+    {
+        
+    };
+*/
+
+    enum PRIO
+    {
+        HIGH = 0,
+        NORM,
+        LOW,
+
+        NUM_PRIORITIES = 3
+    };
+
+    namespace priv
+    {
+        class JobSystem
+        {
+        private:
+            hsd::Counter counter = {0};
```

Comment:
Private members and variables declared inside member functions have `_` prefix as well, good for differentiating public and private variables

Reply:
hello